File: StorageApi/src/main/kotlin/org/gateway/storageApi/consumerGroup/dto/ConsumerGroupDto.kt
package org.gateway.storageApi.consumerGroup.dto

import org.gateway.storage.consumerGroup.ConsumerMode
import org.gateway.storage.consumerGroup.WiringMode
import org.gateway.storageApi.batterySystem.dto.BatterySystemDto
import org.gateway.storageApi.dto.Dto

class ConsumerGroupDto : Dto() {
    lateinit var name: String
    var voltage: Double = 0.0
    var standard: Boolean = false
    lateinit var mode: ConsumerMode
    lateinit var wiring: WiringMode
    var systems: List<BatterySystemDto> = emptyList()
}

File: internal-frontend/src/app/dto/ConsumerGroupDto.ts
import {Dto} from './Dto';
import {WiringMode} from "./WiringMode";
import {ConsumerMode} from "./ConsumerMode";
import {BatterySystemDto} from "./BatterySystemDto";

export class ConsumerGroupDto extends Dto {
  name: string = '';
  voltage = 0;
  standard = false;
  mode: ConsumerMode = ConsumerMode.None;
  wiring: WiringMode = WiringMode.Unknown;
  systems: BatterySystemDto[] = [];
}
